app/src/main/java/com/concordium/wallet/ui/cis2/lookfornew/TokenDetailsFragment.kt

```diff
@@ -81,9 +82,18 @@ class TokenDetailsFragment : TokensBaseFragment() {
         }
     }
 
-    private fun setOwnership(tokenMetadata: TokenMetadata) {
+    private fun setBalance(token: Token, tokenMetadata: TokenMetadata) {
         if (tokenMetadata.unique) {
+            binding.details.balanceHolder.visibility = View.GONE
             binding.details.ownershipHolder.visibility = View.VISIBLE
+            binding.details.ownership.text =                if (token.totalBalance != BigInteger.ZERO)
+                    getString(R.string.cis_owned)
+                else
+                    getString(R.string.cis_not_owned)
+        } else {
+            binding.details.ownershipHolder.visibility = View.GONE
+            binding.details.balanceHolder.visibility = View.VISIBLE
+            binding.details.balance.text = token.totalBalance.toString()
```

Comment:
This should be formatted using the amount of decimal from the metadata. (I hope we have a function for this / we do this in the overview)